docs/backup/backup-rsync.md

```diff
@@ -0,0 +1,180 @@
+> locally, to/from another host over any remote shell, or to/from a remote
+> rsync daemon. It offers a large number of options that control every aspect
+> of its behavior and permit very flexible specification of the set of files
+> to be copied. It is famous for its delta-transfer algorithm, which reduces
+> the amount of data sent over the network by sending only the differences
+> between the source files and the existing files in the destination. Rsync is
+> widely used for backups and mirroring and as an improved copy command for
+> everyday use.
+
+Rsync's main advantage over other backup/copying methods is it's
+delta-transfer algorithm. It ensures that only the modified parts of certain
+files are copied, thus reducing bandwidth use and the time required to
+complete the transfer operation.
+
+### Preparation
+
+For the purpose of this guide, we will assume that you intend to copy the
+remote contents of the `seafile_data` folder to a local drive, as well as the
+database dump.
+
+The remote `seafile_data` directory will be located at
+`/srv/seafile/haiwen/seafile-data` . The local drive will be mounted
+at `/run/media/foo/seafbackup/`.
+
+We will copy the database dumps from the server to our local directory
+`/run/media/foo/seafbackup/seafile_databases` using a tool of your choosing,
+for example SFTP. The remote `seafile_data` folder will be copied to
+`/run/media/foo/seafbackup/seafile_data_backup/` using rsync.
+
+Before you begin the next steps, Install the `rsync` package via your
+distribution's package manager.
+
+### Backup steps
+
+#### 1. Stop Seafile and Seahub
+
+As a systemd service:
+
+```bash
+systemctl stop seafile && systemctl stop seahub
+```
+
+Or using only the script:
+```bash
+./seahub.sh stop && ./seafile.sh stop
+```
+
+#### 2. Backup the databases
+
+SSH into the seafile/database host and backup the databases. You will be asked
+for the MySQL root password:
+
+```bash
+mysqldump -u root -p --opt ccnet-db > ccnet-db.sql.`date +"%Y-%m-%d-%H-%M-%S"` &&
+mysqldump -u root -p --opt seafile-db > seafile-db.sql.`date +"%Y-%m-%d-%H-%M-%S"` &&
+mysqldump -u root -p --opt seahub-db > seahub-db.sql.`date +"%Y-%m-%d-%H-%M-%S"`
+
+```
+Use SFTP to copy them over to a location of your choosing, for example
+to `/run/media/foo/seafbackup/seafile_databases/` on our local drive.
+
+#### 3. Seafile GC
+
+In order to avoid backing up deleted libraries or data, we shall run seafile
+garbage collection first.
+
+We can do a dry run first, without actually deleting any data. Depending on
+your configuration, running it as the `seafile user` might be preferable:
+
+```bash
+runuser -l seafile -c 'cd /srv/seafile/haiwen/seafile-server-latest && ./seaf-gc.sh --dry-run
+```
+
+To run the real garbage collection, remove the `--dry-run` parameter.
+
+```bash
+runuser -l seafile -c 'cd /srv/seafile/haiwen/seafile-server-latest && ./seaf-gc.sh
+```
+
+#### 4. Rsync data backup
```

Comment:
At this point seafile can be started again.

Reply:
What happens if during the rsync run, but after the database backup someone starts making changes?

Wouldn't the `seafile_data` contents and the backed up database end up being out of sync? Isn't it safer, albeit more inconvenient to let the rsync finish first?

Reply:
Well, I guess FSCK would handle that.

Reply:
As you prefer